cluster/engine.go

```diff
@@ -434,7 +459,7 @@ func (e *Engine) refreshLoop() {
 			failedAttempts++
```

Comment:
I think `failureCount` and `failedAttempts` here should be the same things.

- The refresh loop will contribute in increasing the failed count (so after 2 failed refresh loops, we just need to make a single failing call to mark the node as unhealthy)
- It will be configurable by the same `e.opts.RefreshRetry` - maybe the name is wrong now?